Source/Laser/LaserEnvelope.cpp

```diff
@@ -0,0 +1,62 @@
+#include "LaserEnvelope.H"
+#include <AMReX.H>
+
+using namespace amrex;
+
+void LaserEnvelope::AllocateLaserEnvelope (int lev, const BoxArray& ba, const DistributionMapping& dm, const IntVect& ngA, const bool aux_is_nodal)
+{
+    A_nodal_flag = IntVect::TheNodeVector();
+    A_ncomps = ptr.2
+    AllocInitMultiFab(rho_fp[lev], amrex::convert(ba, A_nodal_flag), dm, A_ncomps, ngA, tag("A_fp"), 0.0_rt);
+}
+
+public:
+LaserEnvelope::LaserEnvelope(AmrCore* amr_core, int ispecies, const std::string& name)
+    : LaserEnvelope(amr_core, ispecies),
+      m_laser_name{name}
+{
+    charge = 1.0;
+    mass = std::numeric_limits<Real>::max();
```

Comment:
As discussed offline, these charge and mass parameters do not need to be copied.